src/main/java/chess/domain/ChessGameService.java

```diff
@@ -0,0 +1,104 @@
+package chess.domain;
+
+import chess.db.BoardDao;
+import chess.db.DBBoardRepository;
+import chess.db.DBConnectionUtils;
+import chess.db.GameDao;
+import chess.domain.board.ChessBoardService;
+import chess.domain.board.DefaultBoardInitializer;
+import chess.domain.position.Position;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Map;
+
+public class ChessGameService {
+
+    private static final Color START_COLOR = Color.WHITE;
+    private static final Connection connection = DBConnectionUtils.getConnection();
+
+    private final ChessBoardService chessBoardService;
+    private final ScoreCalculator scoreCalculator;
+    private final GameDao gameDao;
+
+    public ChessGameService(ScoreCalculator scoreCalculator) {
+        this.chessBoardService = new ChessBoardService(new DBBoardRepository(new BoardDao(connection)));
+        this.scoreCalculator = scoreCalculator;
+        this.gameDao = new GameDao(connection);
+    }
+
+    public void initNewGame() {
+        chessBoardService.initNewBoard(DefaultBoardInitializer.initializer());
+        gameDao.setTurn(START_COLOR);
+    }
+
+    public void handleMove(Position from, Position to) {
+        List<Position> movablePositions = chessBoardService.generateMovablePositions(from, gameDao.getCurrentTurn());
+        movePiece(movablePositions, from, to);
+        handleTurn();
+    }
+
+    public void movePiece(List<Position> movablePositions, Position from, Position to) {
+        if (movablePositions.contains(to)) {
+            chessBoardService.movePiece(from, to);
+            return;
+
+        }
+        throw new IllegalArgumentException("해당 기물이 움직일 수 있는 위치가 아닙니다.");
+    }
```

Comment:
```suggestion
    public void movePiece(List<Position> movablePositions, Position from, Position to) {
        if (!movablePositions.contains(to)) {
            throw new IllegalArgumentException("해당 기물이 움직일 수 있는 위치가 아닙니다.");
        }
        chessBoardService.movePiece(from, to);
    }
```
제가 리뷰어에게 받았던 리뷰 공유해 드립니다!
[#640 (comment)](https://github.com/woowacourse/java-chess/pull/640#discussion_r1536842416)
특정 조건에서만 예외가 발생된다면 예외를 발생시키는 부분이 조건문 내부에 위치하는 것이 더 안정적일 것 같아요.